osu.Game.Tests/Visual/Online/TestSceneOverlayView.cs

```diff
@@ -0,0 +1,110 @@
+            AddUntilStep("placeholder is visible", () => view.InternalContents.OfType<LoginPlaceholder>().First().IsPresent);
+            AddUntilStep("spinner is hidden", () => !view.InternalContents.OfType<LoadingSpinner>().First().IsPresent);
+        }
+
+        [Test]
+        public void TestConnectingStatus()
+        {
+            AddStep("set status to connecting", () => ((DummyAPIAccess)API).State = APIState.Connecting);
+            AddUntilStep("placeholder is hidden", () => !view.InternalContents.OfType<LoginPlaceholder>().First().IsPresent);
+            AddUntilStep("spinner is visible", () => view.InternalContents.OfType<LoadingSpinner>().First().IsPresent);
+        }
+
+        [Test]
+        public void TestFailingStatus()
+        {
+            AddStep("set status to failing", () => ((DummyAPIAccess)API).State = APIState.Failing);
+            AddUntilStep("placeholder is hidden", () => !view.InternalContents.OfType<LoginPlaceholder>().First().IsPresent);
+            AddUntilStep("spinner is visible", () => view.InternalContents.OfType<LoadingSpinner>().First().IsPresent);
+        }
+
+        [Test]
+        public void TestOnlineStatus()
+        {
+            AddStep("set status to online", () => ((DummyAPIAccess)API).State = APIState.Online);
+            AddUntilStep("placeholder is hidden", () => !view.InternalContents.OfType<LoginPlaceholder>().First().IsPresent);
+            AddUntilStep("spinner is hidden", () => !view.InternalContents.OfType<LoadingSpinner>().First().IsPresent);
+        }
+
+        private class TestOverlayView : OverlayView<DummyData>
+        {
+            public IEnumerable<Drawable> InternalContents => InternalChildren;
+
+            public TestOverlayView()
+            {
+                Child = new Container
+                {
+                    RelativeSizeAxes = Axes.X,
+                    Height = 400,
+                    Children = new Drawable[]
+                    {
+                        new Box
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Colour = Color4.Blue
+                        },
+                        new OsuSpriteText
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Font = OsuFont.GetFont(Typeface.Torus, 32),
+                            Text = "View content"
+                        },
+                    }
+                };
+            }
+
+            protected override APIRequest<DummyData> CreateRequest() => new DummyAPIRequest();
+
+            protected override void OnSuccess(DummyData response)
+            {
+            }
+
+            private class DummyAPIRequest : APIRequest<DummyData>
+            {
+                protected override string Target => "";
+            }
+        }
+
+        //ReSharper disable once ClassNeverInstantiated.Local
```

Comment:
```suggestion
        [UsedImplicitly]
```
would also work, although I'm not 100% sure this class is worth having (might as well an existing request if we don't care what it is).

Reply:
a dummy class is needed there as `OverlayView<T>` expects a data class to be passed as a generic parameter. I was thinking of creating a tests-wide `DummyAPIData` as well as its corresponding `APIRequest` for cases like this.

Reply:
I know what the thought process was behind adding this. I just think it's unnecessary and you might as well use one of the existing request types instead of creating dummies that do nothing.